source/hardware/controllers/native_instruments_traktor_kontrol_s4_mk3.rst

```diff
@@ -0,0 +1,404 @@
+| Reverse          | - Toggle reverse mode                                            | - Steady on: Flux                        |
+|                  | - On shift, set a loop in the current track position if no active|   enabled                                |
+|                  |   loop, enable loop out wheel mode otherwise                     | - Steady on while pressing               |
+|                  |                                                                  |   :hwlabel:`SHIFT`: loop is active       |
+|                  |                                                                  | - Blinking, with reverse steady on       |
+|                  |                                                                  |   while pressing :hwlabel:`SHIFT`: loop  |
+|                  |                                                                  |   out wheel mode active                  |
++------------------+------------------------------------------------------------------+------------------------------------------+
+| Grid             | - Set the beatgrid at the current track position (on release,    | - On when over a detected beat           |
+|                  |   short press)                                                   | - Blinking when grid/BPM move is on      |
+|                  | - Enable grid move mode while pressed                            | - Blinking when grid move mode is        |
+|                  | - Enable BPM move mode while pressed and pressing                |   enabled                                |
+|                  |   :hwlabel:`SHIFT`                                               |   enabled                                |
++------------------+------------------------------------------------------------------+------------------------------------------+
+| Turntable        | - Toggle on or off the turntable mode                            | - On: Turntable mode on, otherwise jog   |
+|                  |                                                                  |   or vinyl                               |
++------------------+------------------------------------------------------------------+------------------------------------------+
+| Jog              | - Toggle on or off the vinyl mode                                | - On: Vinyl mode on, otherwise jog or    |
+|                  |                                                                  |   turntable                              |
++------------------+------------------------------------------------------------------+------------------------------------------+
+| Deck Select      | - Select a deck                                                  | - The deck's main color will be the one  |
+|                  |                                                                  |   of the selected deck                   |
```

Comment:
In the mapping you use hardcoded colors. AFAIK there is no way to read the deck colors.